File: problems/485_Max_Consecutive_Ones/FullOvellas/Java/1/Solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        var max = 0;
        var current = 0;

        for (var num : nums) {
            if (num == 0) {
                if (current > max) {
                    max = current;
                }
                current = 0;
            }
            current += num;
        }

        if (current > max) {
            return current;
        }
        return max;
    }
}

File: problems/485_Max_Consecutive_Ones/FullOvellas/Rust/1/solution.rs
impl Solution {
    pub fn find_max_consecutive_ones(nums: Vec<i32>) -> i32 {
        let mut curr_consec = 0;
        let mut max_consec = 0;

        
        for num in &nums {
            if *num == 0 {
                max_consec = max_consec.max(curr_consec);
                curr_consec = 0;
            } else {
                curr_consec += 1;
            }
        }
        
        max_consec.max(curr_consec)
    }
}
